src/zombies.test.ts

```diff
@@ -1,30 +1,121 @@
-import { ok } from "node:assert/strict";
+import { ok, equal } from "node:assert/strict";
 import { test } from "node:test";
 
+const numIsValidInteger = (num: unknown): num is number => {
+  return typeof num === 'number' && Number.isInteger(num) && num >= 0;
+};
+
+const isValidString = (name: string): name is string => {
+  return typeof name === 'string' && name.length > 0;
+};
+
+type ZombieReturnObj = {
+  isFull: () => boolean;
+  addZombie: (zombie: string) => void;
+  zombiesInRoom: () => string[];
+  noZombiesInRoom: () => number;
+  spaceLeft: () => number; 
+};
+
 const createRoom = (capacity: number) => {
-  const _capacity = capacity;
+  if (!numIsValidInteger(capacity)) return `Error: Parameter 'capacity' must be a positive integer`;
+  
+  const _capacity = capacity; 
+  let _zombiesInRoom: string[] = [];
 
   return {
-    isFull: () => true,
+    isFull: () => {
+      if (_capacity === 0 || _capacity === _zombiesInRoom.length) return true;      
+      
+      return false;
+    },
+    addZombie: (zombie: string) => {
+      if (isValidString(zombie)) {
+        if (_capacity === _zombiesInRoom.length) {
+          _zombiesInRoom.splice(0, 1, zombie);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice says that you replace the element at the given position. That means that your solution works for these tests, but would break if a second zombie gets eaten. The new zombie should always be added to the end of the list.

Reply:
Still, nice job using splice!

Reply:
Great point!